File: app/src/main/java/ru/dyatel/tsuschedule/data/LessonTable.java
package ru.dyatel.tsuschedule.data;

import android.provider.BaseColumns;

public class LessonTable implements BaseColumns {

    private LessonTable() {
    }

    public static final String PARITY = "parity";
    public static final String WEEKDAY = "weekday";
    public static final String TIME = "time";

    public static final String DISCIPLINE = "discipline";
    public static final String AUDITORY = "auditory";
    public static final String TEACHER = "teacher";

    public static final String TYPE = "type";
    public static final String SUBGROUP = "subgroup";

    public static String getCreateQuery(String name) {
        return "CREATE TABLE " + name + " (" +
                _ID + " INTEGER PRIMARY KEY AUTOINCREMENT," +
                PARITY + " TEXT," +
                WEEKDAY + " TEXT," +
                TIME + " TEXT," +
                DISCIPLINE + " TEXT," +
                AUDITORY + " TEXT," +
                TEACHER + " TEXT," +
                TYPE + " TEXT," +
                SUBGROUP + " CHAR(1)" +
                ")";
    }

    public static String getDropQuery(String name) {
        return "DROP TABLE IF EXISTS " + name;
    }

}

File: app/src/main/kotlin/ru/dyatel/tsuschedule/data/LessonTable.kt
package ru.dyatel.tsuschedule.data

import android.provider.BaseColumns
import android.provider.BaseColumns._ID
import org.jetbrains.anko.db.AUTOINCREMENT
import org.jetbrains.anko.db.INTEGER
import org.jetbrains.anko.db.PRIMARY_KEY
import org.jetbrains.anko.db.TEXT
import org.jetbrains.anko.db.plus

object LessonTable : BaseColumns {

    const val PARITY = "parity"
    const val WEEKDAY = "weekday"
    const val TIME = "time"

    const val DISCIPLINE = "discipline"
    const val AUDITORY = "auditory"
    const val TEACHER = "teacher"

    const val TYPE = "type"
    const val SUBGROUP = "subgroup"

    val columns = arrayOf(
            _ID to INTEGER + PRIMARY_KEY + AUTOINCREMENT,
            PARITY to TEXT,
            WEEKDAY to TEXT,
            TIME to TEXT,
            DISCIPLINE to TEXT,
            AUDITORY to TEXT,
            TEACHER to TEXT,
            TYPE to TEXT,
            SUBGROUP to TEXT
    )

}
